build: make master `v23.2.0-alpha.00000000`

renatolabs opened the pull request:
v23.1.1 has been released, `master` should now be considered to be 23.2.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/103551)
<!-- Reviewable:end -->

Commit:
build: make master `v23.2.0-alpha.00000000`

v23.1.1 has been released, `master` should now be considered to be
23.2.

Epic: none

Release note: None
